core/baseclass.lua

```diff
@@ -24,10 +24,21 @@ SILE.setCommandDefaults = function (command, defaults)
 end
 
 SILE.doTexlike = function (doc)
-  doc = "\\begin{document}" .. doc .. "\\end{document}"
+  -- Setup new "fake" file in which the doc exists
   local cpf = SILE.currentlyProcessingFile
-  SILE.currentlyProcessingFile = "<"..debug.getinfo(2).short_src..":"..debug.getinfo(2).linedefined..">"
-  SILE.process(SILE.inputs.TeXlike.docToTree(doc))
+  local caller = debug.getinfo(2, "Sl")
+  local temporaryFile = "<"..caller.short_src..":"..caller.currentline..">"
```

Comment:
Why overload the concept of "files" instead of just having a status flag of some kind to track when we are in what processing mode? The concept of faking out the file name doesn't quite sit right with me.

Reply:
I am not sure what you mean by this. The `currentlyProcessingFile` change has been there already, but that system can't be reliably used for reporting in which file something was `\defined`, see my [previous comment](https://github.com/simoncozens/sile/pull/603#issuecomment-489264488).

If you refer to the `file` field added to the AST, I don't think that is overloading, as it literally is the file in which the AST node has been defined.

Can you please elaborate?

Reply:
I see this is a more systemic issue in SILE already and not really introduced my this PR. I'll consider addressing it another time, this PR is already hard to follow and shouldn't be tasked with fixing unrelated problems.